### PR TITLE
 feat(save): Save a dataset with multiple file arguments

### PR DESCRIPTION
**Pull request by dustmop**

### CODE REVIEW

**Comment by dustmop** [comment]
Fixes https://github.com/qri-io/qri/issues/690.

File: lib/file.go [hunk 1]
```diff
@@ -58,51 +58,77 @@ func pathKind(path string) string {
...
 }
 
-// ReadDatasetFile decodes a dataset document into a Dataset
-func ReadDatasetFile(path string) (ds *dataset.Dataset, err error) {
-	var (
-		resp *http.Response
-		f    *os.File
-		data []byte
-	)
+// ReadDatasetFiles decodes a dataset document into a Dataset
```
Comment (b5) [inline lib/file.go]: Outdated comment. It'd be great if this function comment broke down some of the nuances of reading multiple filepaths into a dataset, keeping it as high-level as possible. Something like this (adjusting for accuracy):

```go
// ReadDatasetFiles combines resources at zero or more filepaths, deserializing them into a
// *dataset.Dataset. Files can separately specify different components of a dataset
// but two provided components cannot overlap (modify the same component)
```

File: lib/file.go [hunk 2]
```diff
@@ -58,51 +58,77 @@ func pathKind(path string) string {
...
+func ReadDatasetFiles(pathList []string) (*dataset.Dataset, error) {
```
Comment (b5) [inline lib/file.go]: I'd prefer 
```go
func ReadDatasetFiles(pathList ... string) (*dataset.Dataset, error)
```
To me ReadDatasetFiles would be very useful when using qri-as-a-library by hiding the `[]string` from the caller:
```go
//  variadic arguments allows calls for easier single-file loading:
ds, err := lib.ReadDatasetFiles("./path.json")

//  and chaining that doesn't require []string allocation:
ds, err := lib.ReadDatasetFiles("./structure.json", "./meta.yaml")
```

File: cmd/save.go [hunk 3]
```diff
@@ -57,7 +57,7 @@ commit message and title to the save.`,
 		},
 	}
 
-	cmd.Flags().StringVarP(&o.FilePath, "file", "f", "", "dataset data file (yaml or json)")
+	cmd.Flags().StringSliceVarP(&o.FilePaths, "file", "f", nil, "dataset or component file (yaml or json)")
```
[the remark this answers is not in the document]
Comment (dustmop) [reply cmd/save.go]: Added, along with an all-in-one combination of meta + transform + viz.

File: lib/file.go (hunk 2 above)
Comment (dustmop) [reply lib/file.go]: Good idea, this is much better.

File: lib/file.go (hunk 1 above)
Comment (dustmop) [reply lib/file.go]: Done.

**Pull request merged**